move-to-sdk/test/io/sphere/sdk/facets/SelectFacetTest.java

```diff
@@ -0,0 +1,121 @@
+        assertThat(facet.getLimitedOptions()).containsExactlyElementsOf(OPTIONS);
+    }
+
+    @Test
+    public void createsInstanceWithOptionalValues() throws Exception {
+        final SelectFacet<ProductProjection> facet = SelectFacetBuilder.of(KEY, LABEL, SEARCH_MODEL).build();
+        assertThat(facet.getKey()).isEqualTo(KEY);
+        assertThat(facet.getLabel()).isEqualTo(LABEL);
+        assertThat(facet.getType()).isEqualTo(SELECT);
+        assertThat(facet.getSearchModel()).isEqualTo(SEARCH_MODEL.untyped());
+        assertThat(facet.getFacetResult()).isEmpty();
+        assertThat(facet.getSelectedValues()).isEmpty();
+        assertThat(facet.isMatchingAll()).isFalse();
+        assertThat(facet.isMultiSelect()).isTrue();
+        assertThat(facet.getThreshold()).isEmpty();
+        assertThat(facet.getLimit()).isEmpty();
+        assertThat(facet.isAvailable()).isTrue();
+        assertThat(facet.getAllOptions()).isEmpty();
+        assertThat(facet.getLimitedOptions()).isEmpty();
+    }
+
+    @Test
+    public void createsInstanceWithDifferentSelectedValues() throws Exception {
+        final SelectFacet<ProductProjection> facet = SelectFacetBuilder.of(KEY, LABEL, SEARCH_MODEL).build();
+        assertThat(facet.getSelectedValues()).isEmpty();
+        assertThat(facet.withSelectedValues(SELECTED_VALUE_TWO).getSelectedValues()).containsExactlyElementsOf(SELECTED_VALUE_TWO);
+    }
+
+    @Test
+    public void createsInstanceWithDifferentFacetResult() throws Exception {
+        final SelectFacet<ProductProjection> facet = SelectFacetBuilder.of(KEY, LABEL, SEARCH_MODEL).build();
+        assertThat(facet.getFacetResult()).isEmpty();
+        assertThat(facet.withSearchResult(searchResult()).getFacetResult()).contains(FACET_RESULT_WITH_THREE_TERMS);
+    }
+
+    @SuppressWarnings("unchecked")
+    private PagedSearchResult<ProductProjection> searchResult() {
+        final String json = "{" +
+                "  \"facets\": {" +
+                "    \"categories.id\": {" +
+                "      \"type\": \"terms\"," +
+                "      \"missing\": 5," +
+                "      \"total\": 60," +
+                "      \"other\": 0," +
+                "      \"terms\": [" +
+                "        { " +
+                "          \"term\": \"one\", " +
+                "          \"count\": 30 " +
+                "        }," +
+                "        { " +
+                "          \"term\": \"two\", " +
+                "          \"count\": 20 " +
+                "        }," +
+                "        { " +
+                "          \"term\": \"three\", " +
+                "          \"count\": 10 " +
+                "        }" +
+                "      ]" +
+                "    }" +
+                "  }" +
+                "}";
+        final JsonNode node = Json.parse(json);
+        return SphereJsonUtils.readObject(node, PagedSearchResult.class);
```

Comment:
@schleichardt This is the best way I found to obtain a PagedSearchResult. Is there a better way? It's a bit ugly right now. I also tried using the test double client, but it was worse IMO.